src/plugins/ui_core_views/table_style.ts

```diff
@@ -0,0 +1,185 @@
+
+  handle(cmd: Command) {
+    if (
+      invalidateEvaluationCommands.has(cmd.type) ||
+      (cmd.type === "UPDATE_CELL" && "content" in cmd) ||
+      cmd.type === "EVALUATE_CELLS"
+    ) {
+      this.tableStyles = {};
+      return;
+    }
+    switch (cmd.type) {
+      case "HIDE_COLUMNS_ROWS":
+      case "UNHIDE_COLUMNS_ROWS":
+      case "UNFOLD_HEADER_GROUP":
+      case "FOLD_HEADER_GROUP":
+      case "FOLD_ALL_HEADER_GROUPS":
+      case "UNFOLD_ALL_HEADER_GROUPS":
+      case "UPDATE_TABLE":
+      case "UPDATE_FILTER":
+        delete this.tableStyles[cmd.sheetId];
+        break;
+    }
+  }
+
+  finalize() {
+    for (const sheetId of this.getters.getSheetIds()) {
+      if (!this.tableStyles[sheetId]) {
+        this.tableStyles[sheetId] = {};
+      }
+      for (const table of this.getters.getTables(sheetId)) {
+        if (!this.tableStyles[sheetId][table.id]) {
+          this.tableStyles[sheetId][table.id] = this.computeTableStyle(sheetId, table);
+        }
+      }
+    }
+  }
+
+  getCellTableStyle(position: CellPosition): Style | undefined {
+    const table = this.getters.getTable(position);
+    if (!table) {
+      return undefined;
+    }
+
+    return this.tableStyles[position.sheetId][table.id]().styles[position.col]?.[position.row];
+  }
+
+  getCellTableBorder(position: CellPosition): Border | undefined {
+    const table = this.getters.getTable(position);
+    if (!table) {
+      return undefined;
+    }
+    return this.tableStyles[position.sheetId][table.id]().borders[position.col]?.[position.row];
+  }
+
+  private computeTableStyle(sheetId: UID, table: Table): Lazy<ComputedTableStyle> {
+    return lazy(() => {
+      const { config, numberOfCols, numberOfRows } = this.getTableRuntimeConfig(sheetId, table);
+      const relativeTableStyle = getComputedTableStyle(config, numberOfCols, numberOfRows);
+
+      // Return the style with sheet coordinates instead of tables coordinates
+      const mapping = this.getTableMapping(sheetId, table);
+      const absoluteTableStyle: ComputedTableStyle = { borders: {}, styles: {} };
+      for (let col = 0; col < numberOfCols; col++) {
+        const colInSheet = mapping.colMapping[col];
+        absoluteTableStyle.borders[colInSheet] = {};
+        absoluteTableStyle.styles[colInSheet] = {};
+
+        for (let row = 0; row < numberOfRows; row++) {
+          const rowInSheet = mapping.rowMapping[row];
+          absoluteTableStyle.borders[colInSheet][rowInSheet] = relativeTableStyle.borders[col][row];
+          absoluteTableStyle.styles[colInSheet][rowInSheet] = relativeTableStyle.styles[col][row];
+        }
+      }
+      return absoluteTableStyle;
+    });
+  }
+
+  /**
+   * Get the actual table config that will be used to compute the table style. It is different from the config of the table
+   * because of hidden rows and columns in the sheet. For example remove the hidden rows from config.numberOfHeaders.
```

Comment:
could you reduce the length of those lines (to max ~80 chars) ? When I have two files next to each other in vs code, I can't read the end.